sig-release/members.md

```diff
@@ -0,0 +1,8 @@
+## SIG Release Membership
```

Comment:
I second this. Everybody may join the SIG mailing list and be a "member". Let's use more common naming as "SIG leaders" (if the leadership is meant here).